core/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -182,10 +181,37 @@ public String getUUID() {
      * Returns <code>true</code> if the index has a custom data path
      */
     public boolean hasCustomDataPath() {
-        return NodeEnvironment.hasCustomDataPath(settings);
+        return customDataPath() != null;
     }
 
     /**
+     * Returns the customDataPath for this index, if configured. null o.w.
+     */
+    public String customDataPath() {
+        return settings.get(IndexMetaData.SETTING_DATA_PATH);
+    }
+
+    /**
+     * Returns <code>true</code> iff the given settings indicate that the index
+     * associated with these settings allocates it's shards on a shared
+     * filesystem. Otherwise <code>false</code>.
+     */
+    public boolean isOnSharedFilesystem() {
+        return IndexMetaData.isOnSharedFilesystem(getSettings());
```

Comment:
Yeah, I was doubting about this too, but IndexMetaData.isOnSharedFilesystem is used from places where IndexSettings are not used at the moment and it will take more work to open them up/change it (if we want to). Decided to leave it for now.